tawazi/dag.py

```diff
@@ -512,39 +502,92 @@ def executor(self, **kwargs: Dict[str, Any]) -> "DAGExecution":
         """
         return DAGExecution(self, **kwargs)  # type: ignore
 
-    def __call__(self, *args: Any, twz_nodes: Optional[List[Alias]] = None) -> Any:
+    def _make_subgraph(
+        self,
+        target_nodes: Optional[List[Alias]] = None,
+        exclude_nodes: Optional[List[Alias]] = None,
+    ) -> nx.DiGraph:
+        graph = deepcopy(self.graph_ids)
+
+        if target_nodes is not None:
+            target_ids = self._get_target_ids(target_nodes)
+            graph.subgraph_leaves(target_ids)
+
+        if exclude_nodes is not None:
+            exclude_ids = list(chain(*(self._alias_to_ids(alias) for alias in exclude_nodes)))
+
+            for id_ in exclude_ids:
+                # maybe previously removed by :
+                # 1. not being inside the subgraph
+                # 2. being a successor of an excluded node
+                if id_ in graph:
+                    graph.remove_recursively(id_)
```

Comment:
maybe this should be a private method wdyt ?

Reply:
DiGraphEx is not a exposed via the `__init__.py`

Reply:
nice catch